lib/DiscoveryStreamFeed.jsm

```diff
@@ -331,15 +340,24 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
     }
 
     if (!layout || !layout.layout) {
-      if (this.config.hardcoded_basic_layout) {
+      if (
+        this.config.hardcoded_basic_layout ||
+        Services.prefs.getBoolPref(PREF_HARDCODED_BASIC_LAYOUT, false)
+      ) {
         layout = { lastUpdate: Date.now(), ...basicLayoutResp };
       } else {
         layout = { lastUpdate: Date.now(), ...defaultLayoutResp };
       }
     }
 
-    if (layout.spocs && this.config.spocs_endpoint) {
-      layout.spocs.url = this.config.spocs_endpoint;
+    if (
+      layout.spocs &&
+      (this.config.spocs_endpoint ||
+        Services.prefs.getCharPref(PREF_SPOCS_ENDPOINT, ""))
```

Comment:
Looks like this ordering always prefers the value from hardcoded object.

Reply:
Looking, but looks like a good catch.

Reply:
Ah, I see what's happening, both of these prefs are optional. So `this.config.spocs_endpoint` is also empty by default. Ideally we would change 1 with code and the other with FF pref switch. I think the order is likely irrelevant, but I do think switching it is less confusing and more intuitive so I think I'll switch it.